konrad/cloud.py

```diff
@@ -269,3 +269,58 @@ def update_cloud_profile(self, atmosphere, **kwargs):
         mass = self.calculate_mass_cloud(cloud_fraction_array, z,
                                          self._water_density)
         self.mass_content_of_cloud_liquid_water_in_atmosphere_layer = mass
+
+
+class CloudEnsemble(Cloud):
+    """Wrapper to combine several clouds into a cloud ensemble.
+
+    Warning: For now, overlapping clouds are handled very poorly!
+
+    A cloud ensemble can consist of an arbitrary number of clouds.
+    After its initialization it is handled like a normal `Cloud`:
+
+    >>> cloud1 = HighCloud(...)
+    >>> cloud2 = LowCloud(...)
+    >>> cloud_ensemble = CloudEnsemble(cloud1, cloud2)
+    >>> cloud_ensemble.cloud_area_fraction_in_atmosphere_layer
+
+    """
+    def __init__(self, *args):
+        self.clouds = args
+
+    def __getattr__(self, name):
+        cumulative_properties = (
+            'mass_content_of_cloud_ice_in_atmosphere_layer',
+            'cloud_area_fraction_in_atmosphere_layer',
+            'longwave_optical_thickness_due_to_cloud',
+            'shortwave_optical_thickness_due_to_cloud',
+        )
+
+        attr_list = [getattr(c, name) for c in self.clouds]
+
+        if name in cumulative_properties:
+            if name == 'cloud_area_fraction_in_atmosphere_layer':
+                # TODO: Make a decision on how to handle overlapping ice and
+                #  liquid clouds. Most likely some sort of scaling of the
+                # liquid water content has to be applied.
+                attr = np.max(attr_list, axis=0)
+            else:
+                attr = np.sum(attr_list, axis=0)
+        else:
+            # TODO: Is `mean` the correct operation for non-cumulative props?
+            attr = np.mean(attr_list, axis=0)
```

Comment:
This is probably the best for cloud particle sizes, although I am sure it's not the same as having multiple different sizes.